runtime/type-registry/src/lib.rs

```diff
@@ -44,7 +162,9 @@ impl Debug for TypeId {
     }
 }
 
-static GLOBAL_TYPE_REGISTRY: LazyLock<TypeRegistry> = LazyLock::new(TypeRegistry::default);
+static GLOBAL_TYPE_REGISTRY: TypeRegistry = TypeRegistry {
+    map: Mutex::new(BTreeMap::new()),
```

Comment:
We can build a `const`-compatible `HashMap` now (`HashMap::with_hasher(BuildHasherDefault::new())`).